### DISCUSSION TITLE
How are we supposed to test onDrop?

### DISCUSSION THREAD

**Discussion by Flo-Inno** [xyflow/xyflow#4545] [Q&A]
Hi all,

we are currently introducing unit tests to our project, but we are very unsure how we would trigger / test onDrop events on ReactFlow?
We are using vitest + testing-library/react for creating UI unit tests.

To our onDrop function:

`const onDrop = (event: React.DragEvent) => {
    if (reactFlowInstance !== null) {
      const position = reactFlowInstance.project({
                    x: event.clientX - reactFlowBounds.left,
                    y: event.clientY - reactFlowBounds.top
       })!;
       ..... More logic here
    }
}`
The onDrop event handler is triggered when using fireEvent.drop() method but the problem is that the reactFlowInstance is in this case null, so nothing will be executed inside. When I check our method for setting the reactFlowInstance it is called everytime with real instances and not null.

Is this not possible at the moment (why?)?

Thank you and looking forward for some guidance :)

**Comment by vishwakamala** [comment]
@moklick Thanks for this library.
We were facing one issue with Writing UT for few events and Got stuck in writing UT for onDrop event. Can you guide me how can i achieve this?
